deploy/vagrant/run_aws.sh

```diff
@@ -3,15 +3,25 @@ set -e
 # clean up environment
 rm -rf files
 
-vagrant up --provider=aws --no-provision
-
 # generate key
 if [ ! -f files/id_rsa ]
 then
+    mkdir -p files
     ssh-keygen -f files/id_rsa -t rsa -N ''
-# ssh without password
+    # ssh without password
     cat files/id_rsa.pub |awk '{print $1, $2, "Generated by vagrant"}' > files/authorized_keys2
     cat files/id_rsa.pub |awk '{print $1, $2, "Generated by vagrant"}' > files/authorized_keys
 fi
 
-vagrant provision
\ No newline at end of file
+vagrant up --provider=aws --parallel --no-provision
+# before provision, we have to ensure all machines are up and ssh usable
+vagrant provision
+
+if [[ "$?" == "0" ]]; then
+    HERE=$(dirname $0)
+    master=`tail -n 1 $HERE/files/hosts | cut -d' ' -f1`
+    master=`cat .vagrant/provisioners/ansible/inventory/vagrant_ansible_inventory | grep Master | cut -d' ' -f2 | cut -d'=' -f2`
```

Comment:
Oh I actually meant that `master` is being set twice in succession, is it necessary to do both?